sdcm/scan_operation_thread.py

```diff
@@ -383,7 +383,9 @@ def _compare_output_files(self) -> bool:
                 self.log.debug("Compared output of normal and reversed queries is identical!")
                 result = True
             else:
-                self.log.warning("Normal and reversed queries output differs: \n%s", result.stdout.strip())
+                output = result.stdout.strip()
```

Comment:
Maybe we should print it to a new special log file?

Reply:
Agree. And data writing must be done directly in the command which we define in line 377.
For the moment we load whole diff to the RAM.
Then, knowing that the diff may be huge 2Gb+, it becomes a need to pipe the data to a file directly.

Reply:
@roydahan , @vponomaryov ,
I don't think we should currently print and save an extra log file for it.
No developer ever looked at these errors or asked to have any such full output. Not sure it gives any added value, surely not at the moment. We better only consider it per demand.
@vponomaryov , the diff is actually quite small, not sure why you think a single diff is getting to 2Gb+, at least not in recent master code.
I don't expect any memory issue for an operation that apparently takes about 13ms like - 
```
< t:2022-09-28 14:36:56,703 f:scan_operation_thread.py l:326  c:FullPartitionScanThread p:INFO  > Comparing scan queries output files by: diff -y --suppress-common-lines  /tmp/tmp1q3hu1c1 /tmp/tmpuhunh1i7
< t:2022-09-28 14:36:56,716 f:scan_operation_thread.py l:332  c:FullPartitionScanThread p:WARNING > Normal and reversed queries output differs: 
< t:2022-09-28 14:36:56,716 f:scan_operation_thread.py l:332  c:FullPartitionScanThread p:WARNING > 190497180						      <
```

Reply:
@yarongilor the `2Gb+` value was taken from the bugreport:
```
jenkins failed by error: java.io.IOException: Cannot read more than 2Gib at a time
```
Knowing that in common cases the log is up to 10Mb we may assume that there is significant `flood`.

Personally, I met the `300Mb+` side of the diff here: https://github.com/scylladb/scylla-cluster-tests/issues/6056

And, according to the current logic, whole this diff always goes to the RAM first and only then gets processed.

Reply:
it can be implemented as sending 1 line WARNING/ERROR message/event with a summary as @yarongilor  suggested but without output[200]  in and all other logs sent as DEBUG. bit SCT lig will be split automatically